qi-lib/flow/core/deforest-syntax.rkt

```diff
@@ -0,0 +1,293 @@
+#lang racket/base
+
+(provide fusable-stream-producer
+         fsp-range
+         fsp-default
+
+         fusable-stream-transformer
+         fst-filter
+         fst-map
+         fst-filter-map
+         fst-take
+
+         fusable-stream-consumer
+         fsc-foldr
+         fsc-foldl
+         fsc-list-ref
+         fsc-length
+         fsc-empty?
+         fsc-default
+
+         define-and-register-deforest-pass
+         )
+
+(require syntax/parse
+         "passes.rkt"
+         "strategy.rkt"
+         (for-template racket/base
+                       "passes.rkt"
+                       "strategy.rkt"
+                       "deforest-templates.rkt")
+         (for-syntax racket/base
+                     syntax/parse))
+
+(define-literal-set fs-literals
```

Comment:
Seconding the comments about abbrevations: perhaps instead of an inline comment,
a doc (where?) "Deforestation Vocabulary" that explains the conventions? For
location, a root-level documentation directory could hold things that aren't
part of qi-doc. On consistent vocabulary, see also "Stretching the Glasgow
Haskell Compiler" (Young, Henry, & Ericson, FUNARCH 2023), esp. § 3.2 on
"Ubiquitous language."